optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/solver/ConferenceSchedulingScoreRulesXlsxTest.java

```diff
@@ -0,0 +1,228 @@
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import org.optaplanner.examples.common.persistence.AbstractXlsxSolutionFileIO;
+import org.optaplanner.examples.conferencescheduling.app.ConferenceSchedulingApp;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
+import org.optaplanner.examples.conferencescheduling.domain.Room;
+import org.optaplanner.examples.conferencescheduling.domain.Talk;
+import org.optaplanner.examples.conferencescheduling.domain.Timeslot;
+import org.optaplanner.examples.conferencescheduling.persistence.ConferenceSchedulingXlsxFileIO;
+import org.optaplanner.test.impl.score.buildin.hardsoft.HardSoftScoreVerifier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.optaplanner.examples.common.persistence.AbstractXlsxSolutionFileIO.DAY_FORMATTER;
+import static org.optaplanner.examples.common.persistence.AbstractXlsxSolutionFileIO.TIME_FORMATTER;
+
+public class ConferenceSchedulingScoreRulesXlsxTest {
+
+    private final String testFileName = "testConferenceSchedulingScoreRules.xlsx";
+    private final HardSoftScore unassignedScore = HardSoftScore.ZERO;
+    private final transient Logger logger = LoggerFactory.getLogger(getClass());
+
+    private ConferenceSolution initialSolution;
+    private String currentPackage;
+    private String currentConstraint;
+    private HardSoftScore expectedScore;
+
+    private HardSoftScoreVerifier<ConferenceSolution> scoreVerifier = new HardSoftScoreVerifier<>(
+            SolverFactory.createFromXmlResource(ConferenceSchedulingApp.SOLVER_CONFIG));
+    private testConferenceSchedulingScoreRulesReader testFileReader;
+
+    @Before
+    public void setup() {
+        File testFile = new File(getClass().getResource(testFileName).getFile());
+        try (InputStream in = new BufferedInputStream(new FileInputStream(testFile))) {
+            XSSFWorkbook workbook = new XSSFWorkbook(in);
+            this.initialSolution = new ConferenceSchedulingXlsxFileIO().read(testFile);
+            this.testFileReader = new testConferenceSchedulingScoreRulesReader(workbook);
+        } catch (IOException | RuntimeException e) {
+            throw new IllegalStateException("Failed reading inputSolutionFile ("
+                    + testFile.getName() + ").", e);
+        }
+    }
+
+    @Test
+    public void testRules() {
+        ConferenceSolution currentSheetSolution;
+        while ((currentSheetSolution = testFileReader.nextSolution()) != null) {
+            scoreVerifier.assertHardWeight(currentPackage, currentConstraint, expectedScore.getHardScore(), currentSheetSolution);
+            scoreVerifier.assertSoftWeight(currentPackage, currentConstraint, expectedScore.getSoftScore(), currentSheetSolution);
+        }
```

Comment:
Just a suggestion, not a merge blocker: the test could be converted into parameterized test instead of using the while loop. Each sheet represented by one parameter. Advantages:
- one sheet failure won't block testing of the others
- no need to log the sheet name

Reply:
Yes, I was thinking this too. It would give a more fine grained JUnit report, especially if one test fails. For a simple use case that shows how it's done, see LookUpStrategyImmutableTest